### PR TITLE
Add Img component with skeleton + fallback support

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces a new component named `Img`, which enhances image handling by managing loading states and fallbacks. It also includes a corresponding Storybook entry to showcase various use cases for the `Img` component.

### Detailed summary
- Added `Img` component to handle image loading states and fallbacks.
- Implemented state management for image loading (`pending`, `fallback`, `loaded`).
- Created default skeleton and fallback UI elements.
- Developed Storybook stories for `Img` component with various scenarios (e.g., no source, invalid source, custom skeleton/fallback).
- Added a toggle feature to dynamically change the image source in the Storybook demo.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/@/components/blocks/Img.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      {error && (
+        <div
+          className={cn(
+            "fade-in-0 absolute inset-0 overflow-hidden transition-opacity duration-300 [&>*]:h-full [&>*]:w-full",
+            className,
+          )}
+        >
+          {props.fallback || defaultFallback}
+        </div>
+      )}
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/blocks/Img.tsx:63]: should these be explicit states? right now it is possible that both render at the same time, no?

File: apps/dashboard/src/@/components/blocks/Img.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const [error, setError] = useState(false);
+  const { className, ...restProps } = props;
+
+  const showSkeleton = isLoading || !props.src;
+  const defaultSkeleton = <div className="animate-pulse bg-accent" />;
+  const defaultFallback = <div className="bg-muted" />;
+
+  return (
+    <div className="relative">
+      {/* biome-ignore lint/a11y/useAltText: <explanation> */}
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/blocks/Img.tsx:26]: instead allow alt text to be passed in

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/@/components/blocks/Img.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/blocks/Img.tsx:63]: yeah - I'll refactor this, converted to draft - this is not done yet

File: apps/dashboard/src/@/components/blocks/Img.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/blocks/Img.tsx:26]: it is passed in via `restProps`

File: apps/dashboard/src/@/components/blocks/Img.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/blocks/Img.tsx]: updated now

File: apps/dashboard/src/@/components/blocks/Img.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const defaultSkeleton = <div className="animate-pulse bg-accent" />;
+  const defaultFallback = <div className="bg-muted" />;
+
+  return (
+    <div className="relative">
+      <img
+        {...restProps}
+        onLoad={() => {
+          setStatus("loaded");
+        }}
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/blocks/Img.tsx:30]: this fires _every_ time even if an image is already in browser cache? Just making sure the behavior matches

**Review by jnsdls — APPROVED** [review]

File: apps/dashboard/src/@/components/blocks/Img.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+      {status !== "loaded" && (
+        <div
+          className={cn(
+            "fade-in-0 absolute inset-0 overflow-hidden transition-opacity duration-300 [&>*]:h-full [&>*]:w-full",
+            className,
+          )}
+        >
+          {status === "pending" && (props.skeleton || defaultSkeleton)}
+          {(status === "fallback" && props.fallback) || defaultFallback}
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/blocks/Img.tsx:52]: incorrect (or at least inconsistent) && || mixing

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/@/components/blocks/Img.tsx (hunk 4 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/blocks/Img.tsx]: good catch, fixed

**Review by jnsdls — APPROVED** [review]

File: apps/dashboard/src/@/components/blocks/Img.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/blocks/Img.tsx:30]: Yes, Verified `onLoad` is called even if the image is cached

**Pull request merged**